src/NHSD.BuyingCatalogue.Ordering.Api/Controllers/OrdersController.cs

```diff
@@ -226,5 +226,31 @@ public async Task<ActionResult<CreateOrderResponseModel>> CreateOrderAsync([From
             createOrderResponse.Errors = result.Errors.Select(x => new ErrorModel(x.Id, x.Field));
             return BadRequest(createOrderResponse);
         }
+
+        [HttpDelete]
+        [Route("{orderId}")]
+        [Authorize(Policy = PolicyName.CanManageOrders)]
+        public async Task<ActionResult> DeleteOrderAsync(string orderId)
+        {
+            var order = await _orderRepository.GetOrderByIdAsync(orderId);
+
+            if (order is null || order.IsDeleted)
+            {
+                return NotFound();
```

Comment:
I would like to suggest here that if the order has already been deleted the request was successful anyway. So I think it would be better to return NoContent, what do you think? I believe there tends to be a bit of debate around this subject in REST terms and that the request should be idempotent, this way the client doesn't have to cater for the NotFound case & it can be called multiple times & will result in the same response.

Reply:
I don't think that's a particularly cool idea. If we ended up switching to a hard-delete for some reason, we'd then have a change in behaviour which would be mega-weird

Reply:
I'm not sure how it would change with a hard delete. It would result in the same response (200 / 204). The record no longer exists therefore it has been deleted. This discussion does come up a lot from various "REST" talks, maybe a second opinion would help?

Reply:
So if we try and delete an order that has never existed, we would get NoContent? I'm not sure that makes sense. Let's get that 2nd opinion! @bjss-chris

Reply:
Hmm, thanks @CallumBradburyBJSS! This is a slightly tricky one. However, I think I'm leaning towards a 404 not found response.

In this case, I'm not sure the returned status code really affects the operation's idempotency. The order can only be deleted once and assuming that the implementation doesn't randomly start picking off other orders (I've not checked but I'm sure it won't!), i.e. it does nothing, then we can probably argue that the operation is idempotent. I guess it may depend on your definition of idempotent but I tend to think about state when considering whether an operation is idempotent.

[RFC 7321](https://tools.ietf.org/html/rfc7231#section-4.3.5) also states that

> If a DELETE method is successfully applied, the origin server SHOULD send... a 204 (No Content) status code if the action has been enacted and no further information is to be supplied, or a 200 (OK) status code if the action has been enacted and the response message includes a representation describing the status.

If we're not actually deleting anything have we enacted the action? My initial thought is: probably not.

However, I can also understand some of the arguments for returning the same response regardless. For the reasons I've tried to outline above, I think I would initially be tempted to go with the 404 though.

One of the issues here is, of course, the lack of an agreed/adopted standard for RESTful APIs. If one was agreed it would hopefully make some of these choices easier.

Whichever status code we decide to implement now it shouldn't be a huge amount of work to swap over later should it (if we had a reason to do so)?

Does that help at all? (I suspect not!)

Reply:
Cheers Chris! I think if the FE accidentally sent the wrong ID, and it was for a non-existent order, it's better for them to be told the order doesn't exist, so they know they messed up, rather than being told everything was deleted successfully and assuming all is well.

Reply:
All good points 👍  We can go with the 404 then.